composer.json

```diff
@@ -11,10 +11,12 @@
             "email": "dcramer@gmail.com"
         }
     ],
+    "minimum-stability": "dev",
     "require-dev": {
         "friendsofphp/php-cs-fixer": "~2.1",
         "phpunit/phpunit": "^4.8 || ^5.0",
-        "monolog/monolog": "*"
+        "nokitakaze/testhttpserver": "^0.1.0",
```

Comment:
You must ensure that events really sended. And there is no difference between `curl_exec` and `\\ThirdPartyTransport\\Transport::send_request`. If some third party transport breaks in future version (like Monolog) and so our code breaks our user will not want to hear explanations like "this is not our code broken". They just want to be sure everything is working well.

And yes. We will do coverage for all third transport by sending mock events **through every single one of them** until it means that we must to install ftp-servers, email-servers etc.

> server to return a certain response data

Yes. And there it is? Async and outer curl didn\'t tested at all.